src/mongo/db/namespace_string.h

```diff
@@ -298,7 +299,40 @@ class NamespaceString {
         return coll().startsWith("system.");
     }
     bool isNormalCollection() const {
-        return !isSystem() && !(isLocal() && coll().startsWith("replset."));
+        if (isSystem())
+            return false;
+            
+        if (isLocal() && (coll().startsWith("replset.") || coll() == "startup_log"))
+            return false;
+      
+        if (isConfigDB() 
+            && (coll().startsWith("cache.") || coll().startsWith("localReshardingOperations.")
+                || coll() == "transactions" || coll() == "transaction_coordinators"
+                || coll() == "migrationCoordinators" || coll() == "tenantMigrationDonors"
+                || coll() == "rangeDeletions" || coll() == "rangeDeletionsForRename"
+                || coll() == "tenantMigrationRecipients" || coll() == "external_validation_keys"
+                || coll() == "reshardingOperations" || coll() == "localRenameParticipants"
+                || coll() == "vectorClock" || coll() == "collection_critical_sections"
+                || coll() == "image_collection")) {
+            return false;
+        }
+
+        if (isMongos() || serverGlobalParams.clusterRole == ClusterRole::ConfigServer) {
+            if (isConfigDB() 
+                && (coll() == "chunks" || coll() == "databases"
```

Comment:
On the topic of your last push (to address performance), one thing you can do here is add `_sd` to these strings.  This will create `constexpr StringData` literals which among other things encodes the length of the literal into the view, allowing the comparison to run in constant time for strings of differing length rather than linear time.

```suggestion
                && (coll() == "chunks"_sd || coll() == "databases"_sd
```

Note: No need to actually update your PR.  I'll address these minor points as and if this gets merged, just letting you know this is a thing.

Reply:
[sgolemon](https://github.com/sgolemon)
thanks,  👍，I have learned